projectile.el

```diff
@@ -2098,12 +2098,21 @@ With INVALIDATE-CACHE invalidates the cache first.  With FF-VARIANT set to a
 defun, use that instead of `find-file'.   A typical example of such a defun
 would be `find-file-other-window' or `find-file-other-frame'"
   (interactive "P")
-  (projectile-maybe-invalidate-cache invalidate-cache)
-  (let ((file (projectile-completing-read "Find file: "
-                                          (projectile-current-project-files)))
-        (ff (or ff-variant #'find-file)))
-    (funcall ff (expand-file-name file (projectile-project-root)))
-    (run-hooks 'projectile-find-file-hook)))
+  (let (projectile-cached-project-root-temp)
```

Comment:
I'd just do a check whether I'm in a project or not (we've got some predicate for this) and if I'm not I'd issue switch project and rebind `default-directory` to that project's directory.